mlir/lib/Dialect/Rock/IR/MfmaInsnGroup.cpp

```diff
@@ -205,10 +205,10 @@ auto getMfmaInsnGroupAttrMap = []() {
                      {{0, 0, MFMAPermB::none}}}},
 
                    {{MfmaTypeId::Fp16TyId, 64, 64},
-                    {ROCDL::mfma_f32_32x32x4f16::getOperationName(),
+                    {ROCDL::mfma_f32_32x32x8f16::getOperationName(),
```

Comment:
For the next step, we need to use those reduction instructions much more aggressively. Could you:
1. Give a 1 to 1 comparison for the CK comparison selected xdlops and see the perf change. (I remember you mentioned there's a certain percentage of improvement so just want to make sure.)
2. Amend the change the majority of the instruction selection to reduction xdlops. Once this is done, we'd need to do a full convolution + gemm tuning, and see the impact of performance in both cases.

Reply:
Just to be clear, the tables are  exactly the same. Check [our table ](https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/941/files/149b3329c1078f0d15272aea537dc1c609ee46db#diff-beeb368616f48d4988170968b32377f1187a7fc55962da48a57bb9c8104b2970 )vs [their table](https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/1a0b0e7bec7415a5604da4aec73161ad4b3a8feb/include/ck/tensor_operation/gpu/warp/xdlops_gemm.hpp). 

The main difference is that they specify `MPerXdl` and `mRepeats` while we specify directly `mPerWave` . So they can have that table and decide to "compose" `mPerWave = 32*2` from their tuning loop .

To mirror this, without adding more parameters, we need to change the table to hardcode their tuning choices. I guess I can do that for now, but maybe we can discuss this to understand how we want to move on

Reply:
I don't see an inherent difference here. They look the same to me. Why do you think we need to change our table to mirror their solution?
 - They specify mPerXdlops + mRepeats, and with multiplication, they know mPerWave
 - We specify mPerWave + mPerXdlops, and with a division, we know mRepeats

Reply:
I am referring to this point:
```
Amend the change the majority of the instruction selection to reduction xdlops. Once this is done, we'd need to do a full convolution + gemm tuning, and see the impact of performance in both cases.
```
If we amend the change to use more reduction xdlops, we will have to change our table, which will diverge from the one CK uses.

Reply:
@jerryyin I think the point here is that CK has an extra tuning parameter that they use to distinguish `MPerWave = 64 <=> MPerXdlops = 64, MRepeats = 1` from `MPerWave = 64 <=> MPerXdlops = 32, MRepeats = 2`. We currently can't distinguish those with our tuning parameters.

Reply:
@giuseros Did they actually used this flexibility? I believe the answer is no because in the limited tuning parameter selections that we have looked at together, they'd not compose the same wave size in two ways. They always only have the reduction combination.

@krzysz00 There's the background that CK has removed all broadcast cases and replaced them all-together with reduction cases. In your example, when MPerWave = 64 there's only 1 sane reduction xdlops, which is to use `32x32x8fp16`. In this case we can deduce the repeat to be 2 correctly.

I think the intent here is to test and get comparison data. If we can verify the conclusion as CK does, that reduction is at least as good as broadcast, then there's no reason to change our implementation for now.

Reply:
Re do they use the flexibility? Somehow yes. Because they can have our exact table but they are still able to specify (during tuning) only the reductions. If we want to do that,  we will have to change our table and remove the broadcasts. But this means that we will loose even the opportunity to try them. 

And this maybe touches also your second point: they didn't remove the broadcasts: they only tune for reductions, but all the broadcasts are still there in the table. So if they ever want them, they can always try them (while if we change our table, we won't be able to do this)

Reply:
@giuseros 
> Somehow yes. Because they can have our exact table but they are still able to specify (during tuning) only the reductions.

I don't think they and us are tuning based on the same range. In the tuning range they have, I remember they wouldn't pick non-reduction xdlops. If I'm wrong, please refence the line where they have this sort of flexibility.

> they didn't remove the broadcasts: they only tune for reductions

I think their blockwise copy only have reduction flavored the logic because it help to simplify the implementation.

Look, taking one step back, I do think we need data to convince each other. Find me a case where broadcast alternative beat reduction and we can mimic what they do. I'm leaning towards not to do it as for now because tuning and re-tuning is such a pain, and I want to avoid it if possible.